src/screens/ContactScreen/index.js

```diff
@@ -128,35 +131,36 @@ const ContactScreen = ({navigation}) => {
           })) || [];
 
         const dataUser = [...followedUsers, ...unfollowedUsers];
-        setUsers(
+        setUsersSearch(
           dataUser?.filter((item) =>
             isAddParticipant ? !existParticipants.includes(item.username) : item
           )
         );
       }
-      setLoading(false);
+      setIsLoadingSearchResult(false);
     } catch (error) {
-      setLoading(false);
+      setIsLoadingSearchResult(false);
     }
   };
 
   const debounced = React.useCallback(
     debounce((changedText) => {
-      handleSearch();
-      setDebouncedText(changedText);
+      handleSearch(changedText);
     }, 1000),
     []
   );
 
   React.useEffect(() => {
-    try {
-      setLoading(true);
-      getDiscoveryUser();
-      setLoading(false);
-    } catch (error) {
-      setLoading(false);
+    if (text.length === 0) {
+      try {
+        setLoading(true);
+        getDiscoveryUser();
+        setLoading(false);
+      } catch (error) {
+        setLoading(false);
+      }
     }
-  }, []);
+  }, [text]);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `debounced` function now takes an argument, which is passed to `handleSearch`. This is a good change as it allows the search text to be passed directly to the search function, rather than relying on the `text` state variable. The useEffect hook has also been updated to call `getDiscoveryUser` only when `text` is empty, which is a sensible change as it avoids unnecessary API calls.



<!-- This is an auto-generated comment by OSS CodeRabbit -->